tools/permission_prompt_test.py

```diff
@@ -52,178 +52,122 @@ def wrap_test(test_method, test_name):
 
 
 class Prompt(object):
-    def __init__(self, deno_exe):
+    def __init__(self, deno_exe, test_types):
         self.deno_exe = deno_exe
+        self.test_types = test_types
 
-    def run(self,
-            arg,
-            bytes_input,
-            allow_read=False,
-            allow_write=False,
-            allow_net=False,
-            allow_env=False,
-            allow_run=False,
-            no_prompt=False):
+    def run(self, arg, bytes_input, args=[]):
         "Returns (return_code, stdout, stderr)."
-        cmd = [self.deno_exe, PERMISSIONS_PROMPT_TEST_TS, arg]
-        if allow_read:
-            cmd.append("--allow-read")
-        if allow_write:
-            cmd.append("--allow-write")
-        if allow_net:
-            cmd.append("--allow-net")
-        if allow_env:
-            cmd.append("--allow-env")
-        if allow_run:
-            cmd.append("--allow-run")
-        if no_prompt:
-            cmd.append("--no-prompt")
+        cmd = [self.deno_exe, PERMISSIONS_PROMPT_TEST_TS, arg] + args
         return tty_capture(cmd, bytes_input)
 
     def warm_up(self):
         # ignore the ts compiling message
         self.run('needsWrite', b'', allow_write=True)
 
-    def test_read_yes(self):
-        sys.stdout.write("test_read_yes ...")
-        code, stdout, stderr = self.run('needsRead', b'y\n')
+    def test(self):
+        for test_type in self.test_types:
+            test_name_base = "test_" + test_type
+            sys.stdout.write(test_name_base + "_allow_flag" + " ... ")
+            self.test_allow_flag(test_type)
+            print green_ok()
+            sys.stdout.write(test_name_base + "_yes_yes" + " ... ")
+            self.test_yes_yes(test_type)
```

Comment:
Why not iterate through all attributes on the class that start with test_

Reply:
No reason. Just didn't feel like it was needed. It does seem like that might be a little more confusing when debugging these tests.